subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/TestMetricsListenerImpl.kt

```diff
@@ -0,0 +1,97 @@
+package com.avito.runner.scheduler.metrics
+
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import com.avito.runner.service.model.TestCase
+import com.avito.runner.service.model.TestCaseRun
+import com.avito.runner.service.worker.device.Device
+import com.avito.time.TimeProvider
+import java.util.concurrent.ConcurrentHashMap
+
+internal class TestMetricsListenerImpl(
+    private val testMetricsSender: TestMetricsSender,
+    private val timeProvider: TimeProvider,
+    private val loggerFactory: LoggerFactory
+) : TestMetricsListener {
+
+    private val logger = loggerFactory.create<TestMetricsListener>()
+
+    private val onDeviceMarks = ConcurrentHashMap<TestKey, Long>()
```

Comment:
`TestKey, Long` - can it be a named model?
We duplicate this `onDeviceMarks: Map<TestKey, Long>` in two classes